File: geo-project/geo-server/src/main/java/com/huanjing/geo/module/project/entity/PackagePlan.java
package com.huanjing.geo.module.project.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("package_plan")
public class PackagePlan {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String packageType;
    private String packageName;
    private BigDecimal standardPrice;
    private Integer serviceMonths;
    private Integer questionPoolSize;
    private Integer coreQuestionCount;
    private Integer biweeklyFrequency;
    private String monthlyReportDepth;
    private String quarterlyReportDepth;
    private String consultantIntensity;
    private String competitorInsightDepth;
    private String mediaDistributionIntensity;
    private String commitmentTargetIntensity;
    private String targetMetricType;
    private BigDecimal targetMetricValue;
    private Integer targetWindowDays;
    private Boolean enabled;
    private Integer sortOrder;
    private String remark;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private List<PackageChannelQuotaConfig> channelQuotaConfigs;
}

File: geo-project/geo-web/src/api/packagePlan.ts
import request from './request'
import type { PackageChannelQuotaConfig, PackagePlan, PageResult, R } from '@/types'

export interface PackageChannelQuotaPayload {
  channelCode: string
  periodType: 'day' | 'week' | 'month' | 'total' | string
  quotaLimit: number
  enabled: boolean
}

export function getEnabledPackagePlans() {
  return request.get<R<PackagePlan[]>>('/package-plans/enabled')
}

export function getAdminPackagePlans(params: {
  current?: number
  size?: number
  keyword?: string
  enabled?: boolean
  audienceType?: 'internal' | 'partner' | string
}) {
  return request.get<R<PageResult<PackagePlan>>>('/admin/package-plans', { params })
}

export function createPackagePlan(data: {
  packageType: string
  packageName: string
  audienceType?: 'internal' | 'partner' | string
  standardPrice: number
  partnerPoints?: number | null
  partnerVisibleConfigJson?: string
  internalDeliveryConfigJson?: string
  serviceMonths: number
  keywordGroupLimit: number
  keywordGroupLimitA: number
  keywordGroupLimitB: number
  keywordGroupLimitC: number
  monthlyReportDepth: string
  quarterlyReportDepth: string
  consultantIntensity: string
  competitorInsightDepth: string
  mediaDistributionIntensity: string
  commitmentTargetIntensity: string
  targetMetricType: string
  targetMetricValue: number
  targetWindowDays: number
  sortOrder: number
  remark?: string
  channelQuotaConfigs?: PackageChannelQuotaPayload[]
}) {
  return request.post<R<PackagePlan>>('/admin/package-plans', data)
}

export function updatePackagePlan(id: number, data: {
  packageName: string
  audienceType?: 'internal' | 'partner' | string
  standardPrice: number
  partnerPoints?: number | null
  partnerVisibleConfigJson?: string
  internalDeliveryConfigJson?: string
  serviceMonths: number
  keywordGroupLimit: number
  keywordGroupLimitA: number
  keywordGroupLimitB: number
  keywordGroupLimitC: number
  monthlyReportDepth: string
  quarterlyReportDepth: string
  consultantIntensity: string
  competitorInsightDepth: string
  mediaDistributionIntensity: string
  commitmentTargetIntensity: string
  targetMetricType: string
  targetMetricValue: number
  targetWindowDays: number
  sortOrder: number
  remark?: string
  channelQuotaConfigs?: PackageChannelQuotaPayload[]
}) {
  return request.put<R<PackagePlan>>(`/admin/package-plans/${id}`, data)
}

export function updatePackagePlanStatus(id: number, enabled: boolean) {
  return request.put<R<void>>(`/admin/package-plans/${id}/status`, { enabled })
}

export function getPackageChannelQuotas(id: number) {
  return request.get<R<PackageChannelQuotaConfig[]>>(`/admin/package-plans/${id}/channel-quotas`)
}

export function updatePackageChannelQuotas(id: number, data: PackageChannelQuotaPayload[]) {
  return request.put<R<PackageChannelQuotaConfig[]>>(`/admin/package-plans/${id}/channel-quotas`, data)
}
